python/python/bystro/proteomics/proteomics_cli.py

```diff
@@ -0,0 +1,161 @@
+"""Provide a CLI for proteomics analysis."""
+import argparse
+from pathlib import Path
+from typing import Any, BinaryIO
+
+import requests
+from bystro.api.cli import DEFAULT_DIR, authenticate, login
+from msgspec import json as mjson
+
+# ruff: noqa: T201
+
+UPLOAD_PROTEIN_ENDPOINT = "/api/jobs/upload_protein/"
+HTTP_STATUS_OK = 200
+ONE_HOUR_IN_SECONDS = 60 * 60
+
+
+def _package_filename(filename: str) -> tuple[str, tuple[str, BinaryIO, str]]:
+    """Wrap filename in a container suitable for upload through the requests library."""
+    filepath = Path(filename)
+    return (
+        "file",
+        (
+            filepath.name,
+            filepath.open("rb"),
+            "application/octet-stream",
+        ),
+    )
+
+
+def upload_proteomics_dataset(args: argparse.Namespace, *, print_result: bool = True) -> dict[str, Any]:
+    """Upload a fragpipe-TMT dataset through the /api/jobs/upload_protein endpoint.
+
+    Parameters
+    ----------
+    args : argparse.Namespace
+        The arguments passed to the command
+    print_result : bool, optional
+        Whether to print the result of the upload operation, by default True.
+
+    Returns
+    -------
+    dict
+        A json response from the endpoint.
```

Comment:
In a future PR, to follow our desire for validation & typed results, could we convert this to a Struct (and deserialize with that as the type)?

Reply:
Sorry, missed this earlier but yes.